optuna/samplers/_nsga2.py

```diff
@@ -327,3 +376,49 @@ def _crowding_distance_sort(population: List[FrozenTrial]) -> None:
 
     population.sort(key=lambda x: manhattan_distances[x.number])
     population.reverse()
+
+
+def _constrained_dominates(
+    trial0: FrozenTrial, trial1: FrozenTrial, directions: Sequence[StudyDirection]
+) -> bool:
+    """Checks constrained-domination.
+
+    A trial x is said to constrained-dominate a trial y, if any of the following conditions is
+    true:
+    1) Trial x is feasible and trial y is not.
+    2) Trial x and y are both infeasible, but solution x has a smaller overall constraint
+    violation.
+    3) Trial x and y are feasible and trial x dominates trial y.
```

Comment:
That makes sense. I moved this description to the docstring for the `constraints_func` argument in the commit https://github.com/optuna/optuna/pull/2175/commits/eda1fd890680d175c8531280251a09d3d811f0e2. Thank you!

Reply:
Sounds good. What I actually meant though was to describe when a trial is _feasible_. I.e. the following condition and that it's not `any`, although it might be clear. Not sure, what do you think?

```python
all(v <= 0 for v in constraints)
```

Reply:
That is, something like 

> If `constraints_func` returns more than one value for a trial, that trial is considered feasible if and only if all values are equal to 0 or smaller.

Reply:
I'm sorry, but I completely misunderstood your comment. I added the suggested sentence to the definition of the feasibility of a single value in the commit https://github.com/optuna/optuna/pull/2175/commits/24734d72b401eea450c128bd1f6359fb5d1b0876.

Reply:
When I read the description again, and I feel the definition of the constrained domination may be too specific. I'm afraid that it can be noise for the users of the original NSGA-II. What do you think?

![image](https://user-images.githubusercontent.com/3255979/104416136-9ceb3080-55b6-11eb-8c66-90f9e313c84c.png)

Reply:
I was first fine with having it documented purely for maintainers, but after seeing your change, I now think that it won't hurts to have it documented publicly. It's well defined and informative to some. It's not a strong opinion so I'm fine either way.

Reply:
> I now think that it won't hurt to have it documented publicly. It's well defined and informative to some.

Thank you for your reply! Then, I'd like to keep the definition.